### PR TITLE
[DEV-11851] Add search modal & results components to storybook

### PR DESCRIPTION
**Pull request by fgyimah**

### CODE REVIEW

File: packages/ui/src/components/StoryCard/StoryCard.tsx [hunk 1]
```diff
@@ -36,14 +37,16 @@ export function StoryCard({
...
                 className,
             )}
         >
             <Link
                 className={twMerge(
                     'aspect-[10/7]',
-                    size === 'large' ? 'w-full md:aspect-[27/17]' : 'w-1/3 hidden md:block',
+                    size === 'large'
+                        ? 'w-full md:aspect-[27/17]'
+                        : 'w-1/3 hidden md:block md:max-w-[200px]',
```
Comment (MohammadxAli) [inline packages/ui/src/components/StoryCard/StoryCard.tsx:49]: One approach that I have seen in many libraries/apps using `twMerge` is to introduce a `cn()` utility which leverages `clsx` or `classNames` within `twMerge`:

```tsx
import { clsx, type ClassValue } from "clsx";
import { twMerge } from "tailwind-merge";

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}
```

Then you can still use the good old way of doing styles like we used to do before using the objects presentations.

```tsx
cn('aspect-[10/7]', { "w-full md:aspect-[27/17]": size === 'large' })
```

Bounce point, it's _shorter_ :)

(Somehow a big deal for a utility which gets reused a lot)

What do you think Francis?

Comment (fgyimah) [reply packages/ui/src/components/StoryCard/StoryCard.tsx:49]: 👍🏾 I do agree, and it keeps it consistent with how we've been doing it earlier, I would leave it for later though as we'll have to change most parts where we use this syntax

File: packages/ui/src/components/SearchModal/SearchModal.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}: Props) {
+    const { ALGOLIA_API_KEY, ALGOLIA_APP_ID, ALGOLIA_INDEX } = algoliaConfig;
+
+    const searchClient = useMemo(
+        () => algoliasearch(ALGOLIA_APP_ID, ALGOLIA_API_KEY),
+        [ALGOLIA_API_KEY, ALGOLIA_APP_ID],
+    );
+
+    return (
+        <Dialog.Root open={isOpen} onOpenChange={onOpenChange}>
```
Comment (oleg-semyonov) [inline packages/ui/src/components/SearchModal/SearchModal.tsx:48]: ❌ 

The modal can't be closed on mobile 😕

**Review by oleg-semyonov — CHANGES_REQUESTED** [review]
Looks good, but I spotted one major issue on mobile

File: packages/ui/src/components/SearchModal/SearchModal.tsx (hunk 2 above)
Comment (fgyimah) [reply packages/ui/src/components/SearchModal/SearchModal.tsx:52]: https://github.com/prezly/theme-kit-js/pull/440/commits/99f78ceb1773d56de7efe17a3fc63ce2a45a4001

File: packages/ui/src/components/SearchModal/components/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                className="w-full border border-gray-200 rounded px-3 py-2"
+                icon={isMd ? MagnifyingGlassIcon : undefined}
+                inputClassName="p-1 focus:outline-transparent focus-within:outline-transparent"
+                name="query"
+                onChange={(event) => setQuery(event.currentTarget.value)}
+                // TODO: add translations
+                placeholder={`Search ${newsroomName}...`}
+                type="search"
+                value={input}
+            />
```
Comment (MohammadxAli) [inline packages/ui/src/components/SearchModal/components/SearchBar.tsx:51]: When search input is focused, there is an ugly blue `outline` around it. Tested in Firefox:

image

Available when element state is `:focus-visible`.

It probably makes sense to apply `focus-visible:outline-0` to `Input` component.

**Review by MohammadxAli — APPROVED** [review]
Well done 👍 Just a small remark that would be nice to address 👇

File: packages/ui/src/components/SearchModal/components/SearchBar.tsx (hunk 3 above)
Comment (fgyimah) [reply packages/ui/src/components/SearchModal/components/SearchBar.tsx]: https://github.com/prezly/theme-kit-js/pull/440/commits/6241cd1541586721a9d8626d1bb92d38dcaa4383

**Pull request merged**